## <center> Q&A Activity Week - 8

### Reading the dataset to python environment

In [1]:
import pandas as pd
Add_data = pd.read_csv('Advertising.csv')

### 1) Explain the dataset

In [2]:
Add_data

Unnamed: 0     TV  Radio  Newspaper  Sales
0             1  230.1   37.8       69.2   22.1
1             2   44.5   39.3       45.1   10.4
2             3   17.2   45.9       69.3    9.3
3             4  151.5   41.3       58.5   18.5
4             5  180.8   10.8       58.4   12.9
..          ...    ...    ...        ...    ...
195         196   38.2    3.7       13.8    7.6
196         197   94.2    4.9        8.1    9.7
197         198  177.0    9.3        6.4   12.8
198         199  283.6   42.0       66.2   25.5
199         200  232.1    8.6        8.7   13.4

[200 rows x 5 columns]

In [3]:
Add_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  200 non-null    int64  
 1   TV          200 non-null    float64
 2   Radio       200 non-null    float64
 3   Newspaper   200 non-null    float64
 4   Sales       200 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 7.9 KB


The given dataset have 200 rows and 5 columns ,all with non-null values.

Columns are named as Unnamed: 0, TV, Radio,Newspaper & Sales where Unnamed column supposedly indicate Serial No.

TV,Radio and Newspaper columns indicate advertisement expenditure (indicated by Question No:3)

### 2) Check the summary statistics and discuss the max, min, avg, median, and percentiles

In [4]:
Add_data.describe(exclude=[int]) 
# excluding Unnamed column(representing serial no)since its evaluation doesn't convey any meaning
#The 50 percentile is the same as the median

TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000
mean   147.042500   23.264000   30.554000   14.022500
std     85.854236   14.846809   21.778621    5.217457
min      0.700000    0.000000    0.300000    1.600000
25%     74.375000    9.975000   12.750000   10.375000
50%    149.750000   22.900000   25.750000   12.900000
75%    218.825000   36.525000   45.100000   17.400000
max    296.400000   49.600000  114.000000   27.000000

2.1 Analysing Maximum value across columns 

In the Tv column ,maximum  value is 296.40 ;for Radio column,maximum  value is 49.60; for Newspaper column,maximum value is 114.00 and for sales column maximum value is 27.00

Ranking media on the basis of maximum advertisement expenditure, Tv occupies the first position,News paper occupies second position and Radio occupies the third position.

2.2 Analysing Minimum value across columns

In the Tv column ,minimum  value is 0.70 ; for Radio column,minimum  value is 0; for Newspaper column,minimum value is 0.30 and for sales column minimum value is 1.60
Comparing minimum advertisement expenditure across columns,Radio comes first(lowest spending), News paper comes second and TV comes third .

2.3 Analysing Average/Mean value across columns

Average value in the TV column is ≈147.04,average value in the Radio column is ≈23.26,average value in the Newspaper column is ≈30.55 and average value in the sales column is ≈14.02 
Comparing mean advertisement expenditure across columns,Tv occupies first position, Newspaper occupies second position and Radio third position.

2.4 Analysing median value across columns

Median value for TV column is ≈149.75,median value in the Radio column is ≈22.90,median value in the Newspaper column is ≈25.75 and median value in the sales column is ≈12.90 
Comparing median advertisement expenditure across columns,Tv occupies first position, Newspaper occupies second position and Radio third position.

 Mean vs Median

Among the columns,only TV had its median greater mean representing a slight negatively skewed distribution,because most of the values lie at the lower tail of the distribution.
All other columns have a slight positively skewed distibution
Mean median difference is maximum for Newspaper ,meaning high values raising the average.

2.5 Analysing percentiles across columns

Percentiles(TV column):

25% of the values are less than or equal to 74.375 OR  75 % of the values are greater than or equal to 74.375; 50 % of the values are less than or equal to 149.75 OR  50 % of the values are greater than or equal to 149.75 (the 50 percentile defines median);75% of the values are less than or equal to 218.825  OR  25 % of the values are greater than or equal to 218.825

Percentiles(Radio column):

25% of the values are less than or equal to 9.975;50 % of the values are less than or equal to 22.90 ;75% of the values are less than or equal to 36.525  

Percentiles(Newspaper column):

25% of the values are less than or equal to 12.75;50 % of the values are less than or equal to 25.75 ;75% of the values are less than or equal to 45.10

Percentiles(Sales column):

25% of the values are less than or equal to 10.375 ;50 % of the values are less than or equal to 12.90 ;75% of the values are less than or equal to 17.40

### 3) whether the same amount was spent for the three advertisements (TV , Radio and Newspaper).

In [5]:
# Stating the hypothesis

H0= Same amount was spent for the three advertisements,µ1=µ2=µ3

Ha= Amount spent for atleast one advertisement differs from the rest,All μ are not equal

In [6]:
# setting the significance level

Considering significance level(α)=0.05 or 5% 

In [7]:
# Determining p value using one way anova (for comparing means from 3 groups)

In [8]:
import numpy as np
from scipy.stats import f_oneway
x=np.array(Add_data['TV'])
y=np.array(Add_data['Radio'])
z=np.array(Add_data['Newspaper'])
f_oneway(x,y,z)

F_onewayResult(statistic=358.8514595342597, pvalue=4.552931539744962e-103)

In [9]:
# comparing p value with significance level(0.05) 

p value: 4.552931539744962e-103 ≈ 4.55 x 10^-103 < 0.05

Rejecting the null hypothesis

In [10]:
#Conclusion

Since p value is less than the selected significance level of 0.05 ,rejecting the null hypothesis which means that the amount spent for atleast one advertisement differs from the rest i.e, All μ are not equal